tgui/packages/tgui/interfaces/Uplink.js

```diff
@@ -104,6 +104,29 @@ const ItemList = (props, context) => {
       disabled,
     };
   });
+  const GetTooltipMessage = (entry_name, is_illegal, are_contents_illegal) => {
+    if (!is_illegal && !are_contents_illegal) {
+      return entry_name;
+    } else {
+      if (is_illegal) {
+        return (
+          <Tooltip content="This product is powered by our latest technology. Please do not let Nanotrasen R&D steal our confidential designs.">
+            <Box inline position="relative" mr={1}>
+              {entry_name}
+            </Box>
+          </Tooltip>
+        );
+      } else {
+        return (
+          <Tooltip content="The catalogue information is labeled as the proudct is implemented with our technology, but this may not be correct. If you're looking for a product with our technology, be careful of purchasing this.">
```

Comment:
I hate this ambiguousness, but this is inevitable because you don't know which item is going to be created by an actual type of creation.